drivers/at24cxxx/include/at24cxxx_params.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_at24cxxx
+ * @{
+ *
+ * @file
+ * @brief       Default configuration for AT24CXXX
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ovgu.de>
+ */
+
+#ifndef AT24CXXX_PARAMS_H
+#define AT24CXXX_PARAMS_H
+
+#include "board.h"
+#include "at24cxxx.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Set default configuration parameters for the AT24CXXX driver
+ * @{
+ */
+#ifndef AT24CXXX_PARAM_I2C
+#define AT24CXXX_PARAM_I2C             (I2C_DEV(0)) /**< I2C bus the EEPROM is connected to */
+#endif
+#ifndef AT24CXXX_PARAM_ADDR
+#define AT24CXXX_PARAM_ADDR            (AT24CXXX_DEV_ADDR_00) /**< I2C address of the EEPROM device */
+#endif
+#ifndef AT24CXXX_PARAM_PIN_WP
+#define AT24CXXX_PARAM_PIN_WP          (GPIO_UNDEF) /**< EEPROM write protect pin */
+#endif
+#ifndef AT24CXXX_PARAM_EEPROM_SIZE
+    #if defined (MODULE_AT24C256) || defined (MODULE_AT24C256C)
+    #define AT24CXXX_PARAM_EEPROM_SIZE     (AT24C256_EEPROM_SIZE) /**< EEPROM memory capacity in bytes */
```

Comment:
I would move all this device specific macro logic in the defines header and use another generic macro here, `AT24CXXX_EEPROM_SIZE` which will be defined in defines.h:
```c
/**
 * @brief AT24C128 16kB memory
 */
#define AT24C128_EEPROM_SIZE      (16384U)

/**
 * @brief AT24C256 32kB memory
 */
#define AT24C256_EEPROM_SIZE      (32768U)

#if defined (MODULE_AT24C256) || defined (MODULE_AT24C256C)
#define AT24CXXX_EEPROM_SIZE     (AT24C256_EEPROM_SIZE)
#elif defined (MODULE_AT24C128)
#define AT24CXXX_EEPROM_SIZE     (AT24C128_EEPROM_SIZE)
#endif
```

Reply:
yes